From 310579a87a289588cf8c45587354a90973978510 Mon Sep 17 00:00:00 2001
From: "Montazeri, Mehrdad" <mmontaze@amazon.com>
Date: Wed, 31 Jan 2018 09:25:08 -0800
Subject: [PATCH 2/2] fix snprintf macro

---
 win32/include/config.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/win32/include/config.h b/win32/include/config.h
index 8d8548e..304a4a9 100644
--- a/win32/include/config.h
+++ b/win32/include/config.h
@@ -117,7 +117,9 @@ typedef int		    intptr_t;
 /* Windows calls these functions something else
  */
 #define strcasecmp   stricmp
+#if defined (_MSC_VER) && (_MSC_VER < 1900)
 #define snprintf    _snprintf
+#endif
 #define strncasecmp  strnicmp
 
 #define MAXHOSTNAMELEN 1024
-- 
2.11.0.windows.3

